lib/srv/db/postgres/activate-user.sql

```diff
@@ -0,0 +1,28 @@
+create or replace procedure teleport_activate_user(username varchar, roles varchar[])
+language plpgsql
+as $$
+declare
+    role_ varchar;
+begin
+    -- If the user already exists and was provisioned by Teleport, reactivate
+    -- it, otherwise provision a new one.
+    if exists (select * from pg_auth_members where roleid = (select oid from pg_roles where rolname = 'teleport-auto-user') and member = (select oid from pg_roles where rolname = username)) then
+        -- If the user has active connections, just use it to avoid messing up
+        -- its existing roles.
+        if exists (select usename from pg_stat_activity where usename = username) then
+          return;
+        end if;
+        -- Otherwise reactivate the user, but first strip if of all roles to
+        -- account for scenarios with left-over roles if database agent crashed
+        -- and failed to cleanup upon session termination.
+        call teleport_deactivate_user(username);
+        execute format('alter user %I with login', username);
+    else
+        execute format('create user %I in role "teleport-auto-user"', username);
+    end if;
+    -- Assign all roles to the created/activated user.
+    foreach role_ in array roles
+    loop
+        execute format('grant %I to %I', role_, username);
+    end loop;
+end;$$;
```

Comment:
I'm wonder about case where the same user already exists in database and overlaps with teleport username.  
I that case it can happen that current internal user permission that overlaps with teleport username will be escalated to the teleport access.

Reply:
Existing user won't receive extra permissions. The "create user" call will fail with "already exists" error so the procedure will end early and "grant" statements won't be executed.

I've also updated this scenario to return an error instead of letting user to log in, which is how I originally thought about handling it in the RFD. So the behavior is:

```
➜  ~ tsh db connect --db-name=postgres local
psql: error: connection to server at "root.gravitational.io" (127.0.0.1), port 3080 failed:
				your Teleport user requires automatic database user provisioning
				but user "roman@goteleport.com" already exists in this PostgreSQL database and is
				not managed by Teleport
ERROR: exit status 2
```